### DISCUSSION TITLE
Filebrowser

### DISCUSSION THREAD

**Discussion by gitbreiti** [seho85/klipper-dgus#40] [Ideas]
Filebrowser to start prints directly from the display would be great.

**Comment by rouvenschandl** [comment]
Thanks for the suggestion—having a file browser to start prints directly from the display would be great in theory. However, a quick note about the Vyper’s DGUS/DWIN display:

- The display does not receive a live video signal from the Pi. It works with predefined image screens where “buttons” are positioned, and it can show simple data coming directly from the printer.
- Because there’s no video stream, dynamically browsing and selecting changing files on the Pi (e.g., from Spoolman or a local folder) isn’t feasible with this type of display. A real file selection UI would require rendering a live interface, which DGUS doesn’t support.
- DGUS screens are primarily designed for static smart home–style interfaces—fixed images and simple datasets (like temperatures). You can theoretically “play” a sequence by feeding images one by one, but it’s very complex, slow to develop, and not truly dynamic.
- Building even a simple, stable layout on DGUS takes a lot of time (weeks to months) because everything is image- and position-based rather than programmatically rendered.

So while we appreciate the idea, the hardware limitations of the DGUS/DWIN display make a true file browser on the screen impractical. If you’re looking for easier file selection, using the web interface (or a connected touchscreen that supports live rendering) is the more realistic path.
